app/helpers/samples_helper.rb

```diff
@@ -165,7 +165,14 @@ def parsed_samples_for_s3_path(s3_path, project_id, host_genome_id)
     s3_prefix = parsed_uri.path.sub(%r{^/(.*?)/?$}, '\1/')
 
     begin
-      entries = S3_CLIENT_LOCAL.list_objects_v2(bucket: s3_bucket_name, prefix: s3_prefix).contents.map(&:key)
+      entries = []
+      S3_CLIENT_LOCAL.list_objects_v2(bucket: s3_bucket_name, prefix: s3_prefix).each do |response|
+        entries += response.contents.map(&:key)
+        if entries.length >= 10_000
```

Comment:
Nice! I wonder, should the 10,000 limit be an app config or a named constant rather than a number hardcoded here?